BlueDBOnDisk/src/main/java/org/bluedb/disk/file/BlueObjectInput.java

```diff
@@ -86,27 +103,29 @@ public T next() {
 			next = nextValidObjectFromFile();
 		}
 		T response = next;
-		lastBytes = nextBytes;
+
+		lastRawBytes = nextRawBytes;
+		lastUnencryptedBytes = nextUnencryptedBytes;
 		next = null;
-		nextBytes = null;
+		nextRawBytes = null;
+		nextUnencryptedBytes = null;
 		return response;
 	}
 
 	public byte[] nextWithoutDeserializing() {
```

Comment:
I agree completely, I was trying to avoid changing the interface here but I don't think this is very readable as-is. I think the second idea you presented actually seems great, I hadn't thought of that and it doesn't remove any functionality from the interface. I'll make that change.